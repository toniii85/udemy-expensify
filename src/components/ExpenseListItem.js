import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

numeral.register("locale", "es", {
  delimiters: {
      thousands: " ",
      decimal: ","
  },
  abbreviations: {
      thousand: "k",
      million: "m",
      billion: "b",
      trillion: "t"
  },
  ordinal : function (number) {
      return number === 1 ? "er" : "o";
  },
  currency: {
      symbol: "â‚¬"
  }
});
numeral.locale("es");

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount / 100).format("0,0[.]00 $")}
      -
      {moment(createdAt).format("DD [de] MMMM YYYY")}
    </p>
  </div>
);

export default ExpenseListItem;