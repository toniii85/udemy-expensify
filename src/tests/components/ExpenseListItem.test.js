import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";
import numeralLocale from "../../config/numeral-locale";

numeralLocale();
test("should render a list item", () => {
  const expense = expenses[0];
  const wrapper = shallow(<ExpenseListItem {...expense} />)
  expect(wrapper).toMatchSnapshot();
});