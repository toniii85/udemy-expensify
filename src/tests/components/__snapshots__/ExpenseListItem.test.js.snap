// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    1,95 â‚¬
    -
    01 de enero 1970
  </p>
</div>
`;
